lib/game.rb

```diff
@@ -0,0 +1,144 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2], [3, 4, 5], [6, 7, 8],
+    [0, 3, 6], [2, 5, 8], [1, 4, 7],
+    [0, 4, 8], [2, 4, 6]
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    # @board.cells.collect{|space| space == "X"}.size.odd? || @board.cells == @board.reset! ? @player_1 : @player_2
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  def over?
+    @board.full? || self.draw? || self.won?
+  end
+
+  # def won?
+  #   # Why is X the winner?
+  #   WIN_COMBINATIONS.any? do |combo|
+  #     if @board.taken?(combo[0]) && @board.cells[combo[0]] == @board.cells[combo[1]] && @board.cells[combo[1]] == @board.cells[combo[2]]
+  #       return combo
+  #     end
+  #   end
+  # end
+  def won?
+
+     WIN_COMBINATIONS.each do |win_combination|
+    # win_combination is a 3 element array of indexes that compose a win, [0,1,2]
+    # grab each index from the win_combination that composes a win.
+    win_index_1 = win_combination[0]
+    win_index_2 = win_combination[1]
+    win_index_3 = win_combination[2]
+
+    position_1 = @board.cells[win_index_1] # load the value of the board at win_index_1
+    position_2 = @board.cells[win_index_2] # load the value of the board at win_index_2
+    position_3 = @board.cells[win_index_3] # load the value of the board at win_index_3
+
+    if position_1 == "X" && position_2 == "X" && position_3 == "X"
+      return win_combination # return the win_combination indexes that won.
+    elsif position_1 == "O" && position_2 == "O" && position_3 == "O"
+      return win_combination
+      end
+   end
+   false
+  end
+
+  def draw?
+    @board.full? && !self.won?
+  end
+
+  def winner
+    if self.won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.